### ISSUE TITLE
Document the actual behavior of node numbers in SU2 format

### ISSUE THREAD

**Issue by btracey** [su2code/SU2#47]
Currently, the mesh file documentation (http://adl-public.stanford.edu/docs/display/SUSQUARED/Mesh+files) says 
"Each line gives the coordinates for a single node followed by its index number. The node index numbers are the indices used for specifying the connectivity information for elements."

and are specified by
0.00000000000000 0.00000000000000 0
0.50000000000000 0.00000000000000 1
1.00000000000000 0.00000000000000 2
0.00000000000000 0.50000000000000 3
0.50000000000000 0.50000000000000 4
1.00000000000000 0.50000000000000 5
0.00000000000000 1.00000000000000 6
0.50000000000000 1.00000000000000 7
1.00000000000000 1.00000000000000 8

These two facts imply that the number at the end specifies the node number of the location. A consequence of that assumption would be that the nodes could be specified in an arbitrary order, for example as,
1.00000000000000 0.00000000000000 2
0.00000000000000 0.00000000000000 0
0.50000000000000 0.00000000000000 1
1.00000000000000 1.00000000000000 8
1.00000000000000 0.50000000000000 5
0.00000000000000 0.50000000000000 3
0.50000000000000 0.50000000000000 4
0.00000000000000 1.00000000000000 6
0.50000000000000 1.00000000000000 7

This, however, is not true. Looking at CPhysicalGeometry::Read_SU2_Format, the node number is not even used in serial; the nodes are assumed to be given in order, and in the .su2 file the node number can be anything or nothing. The number appears to matter if it is MPI with more than one CPU, but here two different index values are needed (the local and the global). This fact is not mentioned at all in the documentation. The documentation should be changed to match the implementation, or the implementation should be changed to match the documentation.

**Comment by hlkline** [comment]
The documentation is now at:
https://github.com/su2code/SU2/wiki/Mesh-File

I would assume the actual behavior is still the same. Brendan, can you please check the documentation to see if you think it is clear now, and update it if not?

**Comment by btracey** [comment]
I think the writing is clear, but inaccurate. The line I quote in the initial report is still there. 

The code does seem to have been updated, and it now lives in CPhysicalGeometry::Read_SU2_Format_Parallel
Here is the relevant code (for 2D I think). Current lines 6100-6104 of geometry_structure.cpp
````
#ifndef HAVE_MPI
              point_line >> Coord_2D[0]; point_line >> Coord_2D[1];
#else
              if (size > SINGLE_NODE) { point_line >> Coord_2D[0]; point_line >> Coord_2D[1]; point_line >> LocalIndex; point_line >> GlobalIndex; }
              else { point_line >> Coord_2D[0]; point_line >> Coord_2D[1]; LocalIndex = iPoint; GlobalIndex = node_count; }
````
In serial, the index is completely ignored. In parallel, both are used. Unlike what the documentation says, the node index is completely meaningless in parallel, and there is no documentation of a needed global index in parallel. 

I don't know what the correct behavior should be. That is a team decision. I think that the behavior should be roughly the same in serial and parallel. Either the local indices matter, or they don't. If they don't matter, they should be removed. Actually implementing this behavior (in either direction), however, could break a lot of people's code, as in either direction mesh files that were previously working could be different. We could also just have different behavior in serial and parallel, but it should be documented as such. 

Thoughts @economon @fpalacios ?

**Comment by economon** [comment]
This was resolved in #728. The trailing index value no longer exists in either the code or the documentation.
